pkg/datapath/loader/loader_test.go

```diff
@@ -215,45 +172,49 @@ func (s *LoaderTestSuite) testCompileFailure(c *C, ep *testutils.TestEndpoint) {
 		}
 	}()
 
-	l := NewLoaderForTest(c)
+	l := newTestLoader(t)
 	timeout := time.Now().Add(contextTimeout)
 	var err error
 	stats := &metrics.SpanStat{}
 	for err == nil && time.Now().Before(timeout) {
-		err = l.compileAndLoad(ctx, ep, getDirs(c), stats)
+		err = l.compileAndLoad(ctx, ep, getEpDirs(ep), stats)
 	}
-	c.Assert(err, NotNil)
+	require.NotNil(t, err)
 }
 
 // TestCompileFailureDefaultEndpoint attempts to compile then cancels the
 // context and ensures that the failure paths may be hit.
-func (s *LoaderTestSuite) TestCompileFailureDefaultEndpoint(c *C) {
-	s.testCompileFailure(c, &ep)
+func TestCompileFailureDefaultEndpoint(t *testing.T) {
+	ep := testutils.NewTestEndpoint()
+	initEndpoint(t, &ep)
+	testCompileFailure(t, &ep)
 }
 
 // TestCompileFailureHostEndpoint is the same as
 // TestCompileFailureDefaultEndpoint, but for the host endpoint.
-func (s *LoaderTestSuite) TestCompileFailureHostEndpoint(c *C) {
-	s.testCompileFailure(c, &hostEp)
+func TestCompileFailureHostEndpoint(t *testing.T) {
```

Comment:
I meant `TestCompileAndLoadHostEndpoint`, clicked the wrong line...

Reply:
> A bit confused how this test passed before if we didn't create `cilium_net` explicitly before

Probably because we have other tests that create it? Not sure :( I think I checked that running the old test also had that error.